File: easy/javascript/0392-is-subsequence.js
/**
 * O(n)
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
  let subIdx = 0;

  for (let i = 0; i < t.length; i++) {
    if (t[i] === s[subIdx]) {
      subIdx++;
    }
  }

  return subIdx === s.length;
};

console.log(isSubsequence("axc", "ahbgdc"));

File: easy/go/0392-is-subsequence.go
func isSubsequence(s string, t string) bool {
   subIdx := 0 

   if len(s) == 0 {
	return true
   }

   if len(s) > len(t) {
	return false
   }

   for i := 0; i < len(t); i++ {
		if t[i] == s[subIdx] {
			subIdx++
			if subIdx == len(s) {
				return true
			}
		}
   }

   return false
}